### ISSUE TITLE
Destructuring syntax in example

### ISSUE THREAD

**Issue by krokofant** [apollographql/apollo-link-state#136]
I adapted the following example but had an issue. The `data`-field is `undefined` when using this destructuring syntax. The only `networkStatus` and `articles` are available to the function body.
```js
const WrappedComponent = graphql(GET_ARTICLES, {
  props: ({ data: { networkStatus, articles } }) => {
    if (data.loading) {
      return { loading: data.loading };
    }

    if (data.error) {
      return { error: data.error };
    }

    return {
      loading: false,
      networkStatus,
      articles,
    };
  },
})(Articles);
```

Is this issue valid or related to some changed ES6/7 syntax?

An alternative would be `({ data, data: { networkStatus, articles } })`

**Comment by steve-a-jones** [comment]
@krokofant - You are correct, this should be updated in the readme.

Perhaps something like.. 

```javascript
const WrappedComponent = graphql(GET_ARTICLES, {
  props: ({ data: { networkStatus, articles, loading, error } }) => {
    if (loading) {
      return { loading };
    }

    if (error) {
      return { error };
    }

    return {
      loading,
      networkStatus,
      articles,
    };
  },
})(Articles);
```

**Comment by lifeiscontent** [comment]
or

``` es6
const WrappedComponent = graphql(GET_ARTICLES, {
  props: ({ data: { networkStatus, articles, ...data } }) => {
    if (data.loading) {
      return { loading: data.loading };
    }

    if (data.error) {
      return { error: data.error };
    }

    return {
      loading: false,
      networkStatus,
      articles,
    };
  },
})(Articles);
```

**Comment by peggyrayzis** [comment]
Closed via #141
